orchestrator/careplancontributor/applaunch/zorgplatform/service.go

```diff
@@ -225,7 +224,159 @@ func (s *Service) registerFhirClientFactory(config Config) {
+		return &user.SessionData{}, fmt.Errorf("get Organization from Bundle (id=%s): %w", encounterId, err)
+	}
+
+	var patientAndPractitionerBundle fhir.Bundle
+	if err = fhirClient.ReadWithContext(ctx, "Patient", &patientAndPractitionerBundle, fhirclient.QueryParam("_include", "Patient:general-practitioner")); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to fetch Patient and Practitioner bundle: %w", err)
+	}
+	var patient fhir.Patient
+	if err := coolfhir.ResourceInBundle(&patientAndPractitionerBundle, coolfhir.EntryHasID(*encounter.Subject.Reference), &patient); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to find Patient resource in Bundle (id=%s): %w", *encounter.Subject.Reference, err)
+	}
+	var practitioner fhir.Practitioner
+	//TODO: The Practitioner has no indetifier set, so we cannot ensure this is the launched Practitioner. Verify with Zorgplatform
+	// if err := coolfhir.ResourceInBundle(&patientAndPractitionerBundle, coolfhir.EntryHasIdentifier(launchContext.Practitioner.Identifier[0]), &practitioner); err != nil {
+	if err := coolfhir.ResourceInBundle(&patientAndPractitionerBundle, coolfhir.EntryIsOfType("Practitioner"), &practitioner); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to find Practitioner resource in Bundle: %w", err)
+	}
+
+	var conditionBundle fhir.Bundle
+	var conditions []map[string]interface{}
+	//TODO: We assume this is in context as the HCP token is workflow-specific. Double-check with Zorgplatform
+	if err = fhirClient.ReadWithContext(ctx, "Condition", &conditionBundle, fhirclient.QueryParam("subject", "Patient/"+*patient.Id)); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to fetch Conditions bundle: %w", err)
+	}
+	if err := coolfhir.ResourcesInBundle(&conditionBundle, coolfhir.EntryIsOfType("Condition"), &conditions); err != nil {
+		return &user.SessionData{}, fmt.Errorf("unable to find Condition resources in Bundle: %w", err)
+	}
+
+	var reasonReferences []fhir.Reference
+	var conditionMap = make(map[string]interface{})
+
+	for _, condition := range conditions {
+		ref := "Condition/magic-" + condition["id"].(string)
+		reasonReferences = append(reasonReferences, fhir.Reference{
+			Type:      to.Ptr("Condition"),
+			Reference: to.Ptr(ref),
+		})
+		conditionMap[ref] = condition
+	}
+
+	// Zorgplatform does not provide a ServiceRequest, so we need to create one based on other resources they do use
+	serviceRequest := &fhir.ServiceRequest{
+		Status: fhir.RequestStatusActive,
+		Identifier: []fhir.Identifier{
+			{
+				System: to.Ptr("https://zorgplatform.online/task/workflow-id"),
+				Value:  &launchContext.WorkflowId,
+			},
+		},
+		ReasonReference: reasonReferences,
+		Subject: fhir.Reference{
+			Reference: to.Ptr("Patient/" + *patient.Id),
+		},
+		//TODO: We need to set the Requester, this is probably only set after the CPC enrolls it to SCP
+		Performer: []fhir.Reference{
```

Comment:
I think we can deduce the Requester from the local environment (`profile.Identities()`), since this SCP instance only serves 1 tenant (at least for now).